From 47c3768861bf92331fa1761ad1e300238869e9e0 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 21 Aug 2023 17:53:05 +0200
Subject: [PATCH 329/954] refactor(st-clock): code size optimization

Re-ordering structures to avoid gaps and minimize data.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git aaa4f5e861bfe99c6f684e4a4bfc73fce671d24a refactor(st-clock): code size optimization
890c6645d2a0b9febad381bdd8813d6aabc82c8c refactor(st-clock): code size optimization]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ia4f66e9b07caeaef2f411a6c4cf9a67064d75e27
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/323783
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/clk/clk-stm32mp13.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/st/clk/clk-stm32mp13.c b/drivers/st/clk/clk-stm32mp13.c
index b37ee2a5a3..88efb71a79 100644
--- a/drivers/st/clk/clk-stm32mp13.c
+++ b/drivers/st/clk/clk-stm32mp13.c
@@ -1694,7 +1694,7 @@ static const struct stm32_clk_pll *clk_st32_pll_data(unsigned int idx)
 }
 
 struct stm32_pll_cfg {
-	int pll_id;
+	uint8_t pll_id;
 };
 
 static unsigned long clk_stm32_pll_recalc_rate(struct stm32_clk_priv *priv,  int id,
@@ -1780,8 +1780,8 @@ static const struct stm32_clk_ops clk_stm32_pll_ops = {
 }
 
 struct clk_stm32_composite_cfg {
-	int gate_id;
-	int div_id;
+	uint8_t gate_id;
+	uint8_t div_id;
 };
 
 static unsigned long clk_stm32_composite_recalc_rate(struct stm32_clk_priv *priv,
-- 
2.53.0

